### PR TITLE
Implemented error card and error boundary

### PR DESCRIPTION
**Pull request by ifahadashraf**

### CODE REVIEW

File: src/components/error-boundary/error-boundary.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { logger } from '../../utils';
```
Comment (InamTaj) [inline src/components/error-boundary/error-boundary.js]: reorganize imports 😑

File: src/components/error-boundary/error-boundary.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import { ErrorCard } from '../ui/error-card';
+
+class ErrorBoundary extends Component {
+  componentDidCatch(error, info) {
+    logger.error(error, info);
```
Comment (InamTaj) [inline src/components/error-boundary/error-boundary.js]: Look into the `logger.error` function, see how many params it expects.

File: src/components/error-boundary/error-boundary.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+      isError,
+      children,
+    } = this.props;
+
+    if(isError) { return <ErrorCard />; }
+    return children;
+  }
+}
+
+export { ErrorBoundary };
```
Comment (InamTaj) [inline src/components/error-boundary/error-boundary.js]: Try `export class`

File: src/components/error-boundary/error-boundary.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+ErrorBoundary.propTypes = {
+  isError: PropTypes.bool,
+  children: PropTypes.array,
```
Comment (InamTaj) [inline src/components/error-boundary/error-boundary.js]: What if it's just a single element/component?

File: src/components/error-boundary/error-boundary.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+  }
+
+  render() {
+    const {
+      isError,
+      children,
+    } = this.props;
+
+    if(isError) { return <ErrorCard />; }
+    return children;
```
Comment (InamTaj) [inline src/components/error-boundary/error-boundary.js]: try using a ternary operator

File: src/components/error-boundary/error-boundary.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+  }
+}
+
+ErrorBoundary.propTypes = {
+  isError: PropTypes.bool,
+  children: PropTypes.node,
```
Comment (InamTaj) [inline src/components/error-boundary/error-boundary.js]: What if it's a single node or multiple nodes? Is there a way to specify multiple shape types for props?

Comment (ifahadashraf) [reply src/components/error-boundary/error-boundary.js]: Node will always accept a valid child whether it's an object or an array.

**Pull request merged**